### PR TITLE
Apply patches needed to support creating images for full disk encrpytion

### PR DESCRIPTION
**Pull request by jsrain**
This PR contains a set of patches which are needed to build with kiwi encrypted images for SUSE ALP. The patches have been tested on the previous ALP prototypes, see e.g.

https://build.opensuse.org/package/show/SUSE:ALP:Source:Standard:Core:0.1/python-kiwi

### CODE REVIEW

**Comment by schaefi** [comment]
@jsrain Thanks much for your effort, this is a nice addition to the encryption story. Can you look at the shell smells and test errors or do you want us to help with that part ?

Thanks

**Comment by jsrain** [comment]
@schaefi Since the code was initial prepared by Olaf K., I would prefer him to have a look. I will point him to this MR (in fact I already did)

**Comment by jsrain** [comment]
@schaefi BTW: I created the PR from several commits; if it helps integrating them, I'm fine creating one for each commit

File: dracut/modules.d/99kiwi-lib/kiwi-luks-lib.sh
Comment (okirch) [reply dracut/modules.d/99kiwi-lib/kiwi-luks-lib.sh:22]: Okay, this is a potential issue in case systemctl show fails for some reason.

Maybe let's put these lines aboce the systemctl line:

    local UnitFileState=
    local SubState=
    eval $(systemctl show  .... )

Comment (okirch) [reply dracut/modules.d/99kiwi-lib/kiwi-luks-lib.sh:22]: See comment above.

File: kiwi/builder/disk.py
Comment (schaefi) [inline kiwi/builder/disk.py:352]: This change here is the reason for the failed unit test. Can you explain why you are changing the location of the keyfile ? Thanks

**Review by schaefi — CHANGES_REQUESTED** [review]
Thanks for pushing the updates. I did a closer look to the code and had some questions. Let me know if this makes sense. Thanks

**Comment by schaefi** [comment]
> I‘m on vacation and will reply in detail when I’m back. Just one word on the keyfile location change; that’s because / is read only on a transactional system.

Ah ok got it, thanks. So I think we can consider to generally change to the location you proposed. I will run some tests in this regard. Thanks for the clarification

**Comment by schaefi** [comment]
You might have seen it already. In addition to the still open questions there is also a merge conflict now. This was caused because we added support for dnf5 which also introduced a schema change. Thus this generated a conflict for the auto generated API code (kiwi/xml_parse.py). To fix this please do the following on your end:

```
git remote add upstream ssh://git@github.com/OSInside/kiwi.git
git fetch --all

tox -e devel

git rebase upstream/master
--> this causes an expected conflict

source .tox/3/bin/activate
touch  kiwi/schema/kiwi.rng
make kiwi/schema/kiwi.rng

git add kiwi/xml_parse.py
git rebase --continue

git push --force
```

Thanks

**Comment by schaefi** [comment]
I'm also happy to give you Olaf, Jiri write access to the kiwi project so you don't have to work from forks. Just let me know what you prefer. Thanks

**Comment by schaefi** [comment]
> @okirch I‘m on vacation and will reply in detail when I’m back.

me too :) will be back in normal mode next week

**Comment by schaefi** [comment]
@okirch @jsrain any updates on this one ? I did a short smoke test on moving the key file to root's home directory and would be ok with the change. Please update incorporate the schema changes as described and update the unit test regarding the key file location change. After that I think we are in a good merge position ;)

Thanks

File: dracut/modules.d/99kiwi-lib/kiwi-luks-lib.sh
Comment (schaefi) [reply dracut/modules.d/99kiwi-lib/kiwi-luks-lib.sh:55]: I double checked in this one and I think we should not call cryptsetup directly. systemd-cryptsetup reads ```/etc/crypttab``` which is created by kiwi. The location for the keyfile, if any, is part of the crypttab which should be sufficient. Unless there are good reasons for the extra call I vote for getting rid of the extra code. Given there are reasons there should at least be an explanatory comment. Thoughts ?

Comment (okirch) [reply dracut/modules.d/99kiwi-lib/kiwi-luks-lib.sh:55]: When I worked on the encrypted VM images for the first ALP prototype, systemd kept asking for a pass phrase despite the existence of a key file; that's why I introduced this if clause. I can go back and re-check (and make sure crypttab has the right keyfile path).

File: kiwi/builder/disk.py
Comment (okirch) [reply kiwi/builder/disk.py:1533]: Given that we're shipping an image with a well-known password, I'm not worried about leaking that to the log. Others may have objections, so I'm find with removing the clear text password from the log message.

**Comment by okirch** [comment]
Today's my first day back to work after vacation; and I'm now looking into the merge conflict.

**Comment by schaefi** [comment]
> Today's my first day back to work after vacation; and I'm now looking into the merge conflict.

great hope you enjoyed your vacation. I proposed a sequence to let you resolve the conflict a few comments above. Hope this suits you

**Comment by okirch** [comment]
@schaefi - How do we take it from here? Shall we keep working on this PR (which is a bit cumbersome because my submits go through my git project -> jiris -> PR), or do you want me to submit a new one with a slightly simpler setup?

**Comment by schaefi** [comment]
> @schaefi - How do we take it from here? Shall we keep working on this PR (which is a bit cumbersome because my submits go through my git project -> jiris -> PR), or do you want me to submit a new one with a slightly simpler setup?

ok I did the following

- Added @okirch and @jsrain to the kiwi project
- Created a new branch and git format'ed the changes you did and created a set of commits as I think they fit
- Let the schema/API creation run
- I think I found a bug with the include of the keyfile and added an extra commit for that part
- Fixed broken and missing unit tests and added them as extra commits
- Refactored a bit the handling of options passed to the luks_device class

I will push the new branch in a minute and create a new pull request from it. You should then be able to just check out the branch and work and push changes as you see fit

I still would like to get rid of the dracut changes, maybe the fix with the keyfile already helps.

This should make it easier for all of us to contribute to the changes and I hope I did not break it ;)

**Comment by schaefi** [comment]
closing on behalf of #2317

**Pull request closed without merging**
